infra/batch/main.tf

```diff
@@ -92,10 +92,10 @@ data "google_iam_policy" "secret_access_function_batch_datasets" {
 
 # Grant permissions to the service account to access the secrets based on the function config
 resource "google_secret_manager_secret_iam_policy" "policy_function_batch_datasets" {
-  for_each = { for x in local.function_batch_datasets_config.secret_environment_variables : x.key => x }
+  for_each = { for x in local.function_batch_process_dataset_config.secret_environment_variables : x.key => x }
```

Comment:
observation: 👁️ This works as the _function_batch_datasets_config_ secrets are a subset of _function_batch_process_dataset_config_ secrets, and both use the same service account. We can keep it like this, but if in the future, _function_batch_datasets_config_ contains secrets not in _function_batch_process_datasets_config_, we will need to modify this solution.